cloudinit/sources/DataSourceWSL.py

```diff
@@ -238,22 +327,66 @@ def check_instance_id(self, sys_cfg) -> bool:
 
     def _get_data(self) -> bool:
         self.vendordata_raw = None
-        seed_dir = cloud_init_data_dir()
+        user_home = find_home()
+        seed_dir = cloud_init_data_dir(user_home)
+        user_data: Optional[Union[dict, bytes]] = {}
+        should_list = False
 
         try:
             self.metadata = load_instance_metadata(
                 seed_dir, self.instance_name
             )
-            file = self.find_user_data_file(seed_dir)
-            self.userdata_raw = cast(
-                str, util.load_binary_file(file.as_posix())
-            )
-            return True
+        except (ValueError, IOError) as err:
+            LOG.error("Unable to load metadata: %s", str(err))
+
+        try:
+            agent_data = load_agent_data(user_home)
+            user_data = load_landscape_data(self.instance_name, user_home)
+            if user_data is None:
+                # Regular user data
+                if seed_dir is None:
+                    raise ValueError
+                file = self.find_user_data_file(seed_dir)
+                if os.path.exists(file.as_posix()):
+                    bin_user_data = util.load_binary_file(file.as_posix())
+                    user_data = util.load_yaml(bin_user_data)
+                    user_data = (
+                        bin_user_data if user_data is None else user_data
+                    )
```

Comment:
Instead of raising the exception here, I'd rather nest the code below under the if-clause. Raising an exception just for skipping a piece of code is a mistake. The datasource should be OK with finding just the agent data with no user data at all, that's not an exceptional situation.

```suggestion
                if seed_dir is not None:
                    file = self.find_user_data_file(seed_dir).as_posix()
                    if os.path.exists(file):
                        bin_user_data = util.load_binary_file(file)
                        user_data = util.load_yaml(bin_user_data)
                        user_data = (
                            bin_user_data if user_data is None else user_data
                        )
```

If you think this is too nested, then we should extract part of it inside a new method.